metadata-ingestion/src/datahub/ingestion/source/fivetran/config.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from dataclasses import dataclass, field as dataclass_field
+from typing import Dict, List, Optional
+
+import pydantic
+from pydantic import Field
+
+from datahub.configuration.common import AllowDenyPattern, ConfigModel
+from datahub.configuration.source_common import DEFAULT_ENV, DatasetSourceConfigMixin
+from datahub.ingestion.source.state.stale_entity_removal_handler import (
+    StaleEntityRemovalSourceReport,
+    StatefulStaleMetadataRemovalConfig,
+)
+from datahub.ingestion.source.state.stateful_ingestion_base import (
+    StatefulIngestionConfigBase,
+)
+from datahub.ingestion.source_config.sql.snowflake import BaseSnowflakeConfig
+
+logger = logging.getLogger(__name__)
+
+
+class Constant:
+    """
+    keys used in fivetran plugin
+    """
+
+    ORCHESTRATOR = "fivetran"
+    # table column name
+    SOURCE_SCHEMA_NAME = "SOURCE_SCHEMA_NAME"
+    SOURCE_TABLE_NAME = "SOURCE_TABLE_NAME"
+    DESTINATION_SCHEMA_NAME = "DESTINATION_SCHEMA_NAME"
+    DESTINATION_TABLE_NAME = "DESTINATION_TABLE_NAME"
+    SYNC_ID = "SYNC_ID"
+    MESSAGE_DATA = "MESSAGE_DATA"
+    TIME_STAMP = "TIME_STAMP"
+    STATUS = "status"
+    USER_ID = "USER_ID"
+    GIVEN_NAME = "GIVEN_NAME"
+    FAMILY_NAME = "FAMILY_NAME"
+    EMAIL = "EMAIL"
+    EMAIL_DISABLED = "EMAIL_DISABLED"
+    VERIFIED = "VERIFIED"
+    CREATED_AT = "CREATED_AT"
+    CONNECTOR_ID = "CONNECTOR_ID"
+    CONNECTOR_NAME = "CONNECTOR_NAME"
+    CONNECTOR_TYPE_ID = "CONNECTOR_TYPE_ID"
+    PAUSED = "PAUSED"
+    SYNC_FREQUENCY = "SYNC_FREQUENCY"
+    DESTINATION_ID = "DESTINATION_ID"
+    CONNECTING_USER_ID = "CONNECTING_USER_ID"
+
+    # Job status constants
+    SUCCESSFUL = "SUCCESSFUL"
+    FAILURE_WITH_TASK = "FAILURE_WITH_TASK"
+    CANCELED = "CANCELED"
+
+
+SUPPORTED_DATA_PLATFORM_MAPPING = {
+    "postgres": "postgres",
+    "snowflake": "snowflake",
+    "mysql": "mysql",
+}
+
+
+class SnowflakeDestinationConfig(BaseSnowflakeConfig):
+    database: str = Field(
+        default=None, description="The fivetran connector log database."
+    )
+    log_schema: Optional[str] = Field(
+        default="FIVETRAN_LOG", description="The fivetran connector log schema."
+    )
+
+
+class FivetranLogConfig(ConfigModel):
+    destination_platform: str = pydantic.Field(
+        default=None,
```

Comment:
Can we set it to default `snowflake` as we are currently supporting only `snowflake`? 
let's also add a validation for valid values of this argument.